lib/std/heap/logging_allocator.zig

```diff
@@ -1,57 +1,57 @@
 const std = @import("../std.zig");
 const Allocator = std.mem.Allocator;
 
-/// This allocator is used in front of another allocator and logs to the provided stream
-/// on every call to the allocator. Stream errors are ignored.
+/// This allocator is used in front of another allocator and logs to the provided writer
+/// on every call to the allocator. Writer errors are ignored.
 /// If https://github.com/ziglang/zig/issues/2586 is implemented, this API can be improved.
-pub fn LoggingAllocator(comptime OutStreamType: type) type {
+pub fn LoggingAllocator(comptime WriterType: type) type {
```

Comment:
This should be changed to use `std.log` facilities, and it no longer needs to be a generic type.

Reply:
Does that mean we won't be able to test the logging output anymore?

Reply:
we can add a "compare output" test case for it: https://github.com/ziglang/zig/blob/master/test/compare_output.zig

Reply:
This gave me another idea:  #5738

Reply:
@andrewrk 
> This should be changed to use std.log facilities

What to use?
`log.notice`, `log.info`, `log.debug`?

What scope? `.loggingAllocator`?

> we can add a "compare output" test case for it

How? With `log` function in `compare_output.zig`?